script/grblcon.py

```diff
@@ -603,6 +651,18 @@ def _open_port(self, portstr='/dev/ttyUSB0', baud=38400):
                 self._serial_thread.start()
             except (serial.serialutil.SerialException, FileNotFoundError):
                 self._log_crit("Unable to open port: {}".format(portstr))
+    
+    def poll_adcs(self,*args):
+        """Periodically polls the ADC values in grbl
+        
+        :returns: None
+        :rtype: None        
+
+        """
+        if self.connected:
+          self._write('|')
+        
+        self.loop.set_alarm_in(self.adc_poll_time, self.poll_adcs)
```

Comment:
I wasn't familiar with the urwid package. Put in a comment explaining that this line will call the function again soon (as opposed to making it an interrupt handler or a callback to some event or something; it wasn't obvious to me what `self.loop.set_alarm_in` did).

...would it make more sense just to make a daemon thread that just runs this loop itself? It's more code, but I think the intention is easier to discern.

```python
def poll_adcs(self):
    """Should be run as a daemon thread."""
    while True:
        time.sleep(self.adc_poll_time)
        self._write('|')

def run(self):
    ...
    adc_poller_daemon = threading.Thread(target=self.poll_adcs)
    adc_poller_daemon.daemon = True
    adc_poller_daemon.start()
```

edit: ...oh, never mind. You're using a raw `serial.Serial` class, as opposed to the `keyme.util.SerialChannel` class available in the kiosk repo. Our version has nice threadsafe properties, while the raw one does not.